test/val/val_id_test.cpp

```diff
@@ -112,6 +114,26 @@ const char kOpenCLMemoryModel64[] = R"(
   ASSERT_EQ(expected, result);                                                 \
   spvContextDestroy(context);
 
+#define CHECK_WITH_DIAGNOSTIC(str, expected, err_substr)                       \
```

Comment:
This is quite a lot of new code, in a macro.

Much better to follow the example in val_ssa_test.cpp 
There each test is just a string constant, then two or three more lines.  The heavy lifting of object management is taken care of in the test fixture base class.